src/main/java/com/aws/greengrass/shadowmanager/sync/strategy/BaseSyncStrategy.java

```diff
@@ -284,23 +283,6 @@ public void putSyncRequest(SyncRequest request) {
             return;
         }
 
-        try {
-            if (!request.isUpdateNecessary(context)) {
-                logger.atDebug()
-                        .addKeyValue(LOG_THING_NAME_KEY, request.getThingName())
-                        .addKeyValue(LOG_SHADOW_NAME_KEY, request.getShadowName())
-                        .addKeyValue("type", request.getClass())
-                        .log("Ignoring sync request since update is not necessary");
-                return;
-            }
-        } catch (SkipSyncRequestException | RetryableException | UnknownShadowException e) {
-            logger.atWarn(SYNC_EVENT_TYPE)
-                    .addKeyValue(LOG_THING_NAME_KEY, request.getThingName())
-                    .addKeyValue(LOG_SHADOW_NAME_KEY, request.getShadowName())
-                    .log("Ignoring sync request since update is not necessary");
-            return;
-        }
-
```

Comment:
nice. Is the `isUpdateNecessary(context)` method ever called from outside of the Request object now? If it isn't, can the interface be changed to make it accessible just to the model package?